aeneas/src/mach/MachLowering.v3

```diff
@@ -1862,6 +1891,38 @@ class MachLowering(mach: MachProgram, compiler: Compiler, config: MachLoweringCo
 		}
 		return loads;
 	}
+	def genNormTypedLoadsByteRegion(source: Source, machType: Type, base: SsaInstr, offset: int) -> Array<SsaInstr> {
+		var tn: TypeNorm;
+		match (machType) {
+			x: IntType => tn = mach.intNorm.makeType(x);
+			_ => return [ptrLoad(machType, base, offset)];
+		}
+
+		var loads = Array<SsaInstr>.new(tn.size);
+		for (i < tn.size) {
```

Comment:
What about factoring out the first `tn.size - 1` into its own loop? (and also, AFAICT, those don't need to be `IntType`, as `mach.sizeOf()` should handle the other cases)